src/screens/HomeScreen/components/NameBarContainer.js

```diff
@@ -25,18 +24,27 @@ const handleClearButtonPress = async () => {
 const NameBarContainer = () => {
   const [name, setName] = useState('');
   const [inputText, setInputText] = useState('');
-  const [userData, setUserData] = useState('');
+  const [userData, setUserData] = useState({});
 
   useEffect(() => {
     getData().then((value) => setName(value));
   }, []);
 
   useEffect(() => {
-    getUserData(name)
-      .then((list) => list.wordList.map((word) => word.word))
-      .then((value) => setUserData(value));
+    getUserWordList(name).then((value) => setUserData(value));
   }, [name]);
 
+  const refreshList = () => {
+    getUserWordList(name).then((value) => setUserData(value));
+  };
+
+  const getWords = () => {
+    const words = {};
+    Object.keys(userData).map(
```

Comment:
For a fun little challenge you could use `.reduce` here instead of `.map` and return the result directly.

```js
const getWords = () => {
  return Object.keys(userData).reduce(..)
}
```